lib/net/http/request.rb

```diff
@@ -1,8 +1,82 @@
 # frozen_string_literal: false
-# HTTP request class.
-# This class wraps together the request header and the request path.
-# You cannot use this class directly. Instead, you should use one of its
-# subclasses: Net::HTTP::Get, Net::HTTP::Post, Net::HTTP::Head.
+
+# This class is the base class for \Net::HTTP request classes;
+# it wraps together the request path and the request headers.
+#
+# The class should not be used directly;
+# instead you should use its subclasses, which are covered in the sections below.
+#
+# == About the Examples
+#
+# Examples here assume that <tt>net/http</tt> has been required
+# (which also requires +uri+):
+#
+#   require 'net/http'
+#
+# Many code examples here use these example websites:
+#
+# - https://jsonplaceholder.typicode.com.
+# - http://example.com.
+#
+# Some examples also assume these variables:
+#
+#   uri = URI('https://jsonplaceholder.typicode.com')
+#   uri.freeze # Examples may not modify.
+#   hostname = uri.hostname # => "jsonplaceholder.typicode.com"
+#   port = uri.port         # => 443
+#
+# An example that needs a modified URI first duplicates +uri+, then modifies:
+#
+#   _uri = uri.dup
+#   _uri.path = '/todos/1'
+#
+# == Requests
+#
+# === \Net::HTTP::Get
+#
+# A GET request may be sent using request class \Net::HTTP::Get:
+#
+#   req = Net::HTTP::Get.new(uri) # => #<Net::HTTP::Get GET>
+#   Net::HTTP.start(hostname) do |http|
+#     http.request(req)
+#   end # => #<Net::HTTPOK 200 OK readbody=true>
+#
+# === \Net::HTTP::Head
+#
+# A HEAD request may be sent using request class \Net::HTTP::Head:
+#
+#   req = Net::HTTP::Head.new(uri) # => #<Net::HTTP::Head HEAD>
+#   Net::HTTP.start(hostname) do |http|
+#     http.request(req)
+#   end # => #<Net::HTTPOK 200 OK readbody=true>
+#
+# === \Net::HTTP::Post
+#
+# A POST request may be sent using request class \Net::HTTP::Post:
+#
+#   json = {title: 'foo', body: 'bar', userId: 1}
+#   # => "{\"title\":\"foo\",\"body\":\"bar\",\"userId\":1}"
+#   _uri = uri.dup
+#   _uri.path = '/posts'
+#   req = Net::HTTP::Post.new(_uri) # => #<Net::HTTP::Post POST>
+#   req.body = json
```

Comment:
`req.body` should be a string rather than a hash I think.

```suggestion
#   _uri = uri.dup
#   _uri.path = '/posts'
#   req = Net::HTTP::Post.new(_uri) # => #<Net::HTTP::Post POST>
#   req.body = '{"title": "foo", "body": "bar", "userId": 1}'
```